### PR TITLE
Add test coverage

### LINKED ISSUE #63: Add testing coverage
- [x] Set up codecov on this repo
- [x] Turn on coverage in the test action
- [x] Upload coverage to codecov
- [x] Add coverage badge to readme

### PR DESCRIPTION
**Pull request by ReubenJ**
Closes #63

### CODE REVIEW

**Comment by THinnerichs** [comment]
Which Julia versions do we want to test for? Shouldn't `nightly` already include 1.10?

**Comment by ReubenJ** [comment]
Nightly is the current state of the master branch, I believe. It's possible that something is broken in nightly that is working in 1.10 just because a new (broken) feature has been merged to master, or something like that.

I think we should be testing 1.8 and 1.10 at least since we say we support >1.8, and 1.10 is the latest stable release. If we want to be more thorough about it, we should test 1.9 and nightly, too, but either/both of those are probably overkill for now.

**Comment by ReubenJ** [comment]
It looks like some other bigger projects, like those from the SciML ecosystem, are just using "1" as a test version: [SciML/DifferentialEquations.jl's CI](https://github.com/SciML/DifferentialEquations.jl/blob/8ce23d22460c66a23944734a86e75209f5b3d264/.github/workflows/CI.yml#L17).

File: .github/workflows/CI.yml [hunk 1]
```diff
@@ -19,6 +19,7 @@ jobs:
       matrix:
         version:
           - '1.8'
+          - '1.10'
```
[the remark this answers is not in the document]
Comment (ReubenJ) [reply .github/workflows/CI.yml:22]: Ha, I just had the same comment. 👍

**Review by THinnerichs — APPROVED** [review]

**Pull request merged**